File: packages/sb-uhq/src/self/commands/fun/spoiler.js
"use strict";

const { messageEdit } = require("../../func/message_edit");

/**
 * @param {import("discord.js-selfbot-v13").Client} client
 * @param {{ channelId: string, text: string }} payload
 */
async function execute(client, payload) {
  const { channelId, text } = payload;
  if (!channelId || !text) throw new Error("channelId et text requis.");
  const channel = await client.channels.fetch(channelId).catch(() => null);
  if (!channel) throw new Error(`Salon ${channelId} introuvable.`);
  const spoilered = `||${text}||`;
  await channel.send(spoilered);
  return { sent: spoilered };
}

// ── Handler commande préfixe ──────────────────────────────────────────────────

async function callback(client, message, args) {
  if (!args.length) return messageEdit(message, "`❌` **Usage :** `.spoiler <texte>`");
  const spoilered = `||${args.join(" ")}||`;
  return message.edit(spoilered).catch(() => message.channel.send(spoilered));
}

module.exports = { name: "spoiler", execute, callback };

File: packages/sb-uhq/app/commands/fun/spoiler.py
"""fun.spoiler — ||texte caché|| (bridge + commande préfixe)."""

from __future__ import annotations

from ...func.discord_util import fetch_channel
from ...func.message_edit import message_edit


async def execute(client, payload):
    channel_id = payload.get("channelId")
    text = payload.get("text")
    if not channel_id or not text:
        raise ValueError("channelId et text requis.")
    channel = await fetch_channel(client, channel_id)
    if not channel:
        raise ValueError(f"Salon {channel_id} introuvable.")
    spoilered = f"||{text}||"
    await channel.send(spoilered)
    return {"sent": spoilered}


async def callback(client, message, args):
    if not args:
        return await message_edit(message, "`❌` **Usage :** `.spoiler <texte>`")
    spoilered = f"||{' '.join(args)}||"
    return await message_edit(message, spoilered)
